li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -771,4 +818,37 @@ private void ensureOpen() {
       throw new IllegalStateException("this federated consumer has already been closed");
     }
   }
+
+  private void maybeResubscribe() {
+    if (_closed || _currentSubscription.getSubscriptionType() == FederatedSubscriptionState.SubscriptionType.NONE ||
```

Comment:
Should we consider fallback mode here? If mario is not available, there is no need to continue the rest of the code. In the future we will have to change the registerFederatedClient API to return a boolean indicating whether it registers itself successfully within the timeout and use fallback mode if not. 

Can you please add a TODO here to check the register state?

Reply:
Added a TODO